out_writeapi_test.go

```diff
@@ -38,3 +44,143 @@ func TestFLBPluginRegister(t *testing.T) {
 	assert.Equal(t, "writeapi", currplugin.name)
 
 }
+
+type StreamChecks struct {
+	calledGetContext     bool
+	calledcheckResponses bool
+	createDecoder        bool
+	gotRecord            bool
+	calledparseMap       bool
+	calledJTB            bool
+	appendRows           bool
+	appendQueue          bool
+}
+
+type MockManagedStream struct {
+	managedstream  *managedwriter.ManagedStream
+	AppendRowsFunc func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error)
+	StreamName     func() string
+	CloseFunc      func() error
+}
+
+func (m *MockManagedStream) AppendRows(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+	return m.AppendRowsFunc(ctx, data, opts...)
+}
+
+func (m *MockManagedStream) Close() error {
+	return m.managedstream.Close()
+}
+
+func TestFLBPluginFlushCtx(t *testing.T) {
+	var checks StreamChecks
+	var setID interface{}
+	var id int
+
+	mockMS := &MockManagedStream{
+		AppendRowsFunc: func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+			checks.appendRows = true
+			return nil, nil
+		},
+	}
+
+	originalFunc := getWriter
+	getWriter = func(client *managedwriter.Client, ctx context.Context, projectID string, opts ...managedwriter.WriterOption) (MWManagedStream, error) {
+		return mockMS, nil
+	}
+	defer func() { getWriter = originalFunc }()
+
+	patch1 := monkey.Patch(output.FLBPluginConfigKey, func(plugin unsafe.Pointer, key string) string {
+		return ""
+	})
+	defer patch1.Unpatch()
+
+	patchDescriptor := monkey.Patch(getDescriptors, func(curr_ctx context.Context, managed_writer_client *managedwriter.Client, project string, dataset string, table string) (protoreflect.MessageDescriptor, *descriptorpb.DescriptorProto) {
+		return nil, nil
+	})
+	defer patchDescriptor.Unpatch()
+
+	patchSetContext := monkey.Patch(output.FLBPluginSetContext, func(plugin unsafe.Pointer, ctx interface{}) {
+		log.Println("Ran set context")
+		setID = ctx
+	})
+	defer patchSetContext.Unpatch()
+
+	plugin := unsafe.Pointer(nil)
+	// ctx := unsafe.Pointer(nil)
+	initRes := FLBPluginInit(plugin)
+
+	orgFunc := getContext
+	getContext = func(ctx unsafe.Pointer) int {
+		checks.calledGetContext = true
+		uintptrValueBack := uintptr(ctx)
+		id = int(uintptrValueBack)
+		return id
```

Comment:
I'm not clear on how you are using ctx here to obtain the id value. Shouldn't you simply return the value that was set in FLBPluginSetContext()?

Reply:
Yeah, true. The Ctx is just the value we set in SetContext but casted to an unsafe.Pointer type and this is converting it back. I guess this is an unnecessary step for the test and I will remove it!

Reply:
Done.